src/main/java/com/microsoft/sqlserver/jdbc/SQLCollation.java

```diff
@@ -602,7 +600,23 @@ final Encoding checkSupported() throws UnsupportedEncodingException
         return this;
     }
 
-    final String charsetName() { return charsetName; }
+    final Charset charset() throws SQLServerException
+    {
+        try
+        {
+            checkSupported();
+            if (charset == null)
+            {
+                charset = Charset.forName(charsetName);
```

Comment:
Yes I considered this. It would be possible but:
'private Charset charset' ist not a local variable but an instance/member variable. `Encoding#charset()` is called quite frequently and by many users both directly and indirectly. If you look at the implementation of `Charset#forName` it is quite involved and the worst case is quite bad, eg. some of the charsets are in the Extended Encoding Set. I would be more comfortable with looking them up just once.